File: A_Alice_and_Bob.py
import sys, math, heapq as heap, itertools
from collections import defaultdict, Counter, deque
from bisect import bisect_right, bisect_left
from random import randint
from heapq import heappush, heappop, heapify


number = lambda: int(sys.stdin.readline().strip())
numbers = lambda: list(map(int, sys.stdin.readline().strip().split()))
words = lambda: sys.stdin.readline().strip().split()
word = lambda: sys.stdin.readline().strip()
yes_no = lambda condition: 'YES' if condition else 'NO'
test_cases = lambda inp=0: number() if not inp else inp


def solve():
    n , m = numbers()
    arr = numbers()
    countless = 0
    countmore = 0
    for i in range(len(arr)):
        if m > arr[i]:
            countless += 1
        elif m < arr[i]:
            countmore += 1
    if countmore >= countless:
        print(m + 1)
    else:
        print(m - 1)
    return

for _ in range(test_cases()):
    solve()

File: A_Alice_and_Bob.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int t;
    cin >> t;
    for (int i = 0; i < t; i++){
        int n, m;
        int l = 0, r = 0;
        cin >> n >> m;
        vector<int> arr(n);
        for (int j = 0 ; j < n; j++){
            cin >> arr[j];
            if (m < arr[j]) ++l;
            if (m > arr[j]) ++r;
        }
        cout << (l <= r ? m + 1 : m - 1) << endl;

    }
    return 0;
}